external/model-preparation-algorithm/mpa_tasks/apis/detection/task.py

```diff
@@ -330,9 +339,7 @@ def _patch_evaluation(config: MPAConfig):
         cfg.metric = 'mAP'
         cfg.save_best = 'mAP'
         # EarlyStoppingHook
-        for cfg in config.get('custom_hooks', []):
-            if 'EarlyStoppingHook' in cfg.type:
-                cfg.metric = 'mAP'
+        config.early_stop_metric = 'mAP'
```

Comment:
To always be equal to the metric of evaluation, looks better to use early_stop_metric = cfg.metric. (The same goes for other tasks)

Reply:
I put the code part in task.py but cls uses mutiple metrics, so early_stop_metric = cfg.metric will bring the lists of metrics in task.py in apis folder.